src/ReasonReactRouter.re

```diff
@@ -67,50 +67,68 @@ let arrayToList = a => {
 /* sigh URLSearchParams doesn't work on IE11, edge16, etc. */
 /* actually you know what, not gonna provide search for now. It's a mess.
    We'll let users roll their own solution/data structure for now */
-let path = () =>
-  switch ([%external window]) {
-  | None => []
-  | Some((window: Dom.window)) =>
-    switch (window |> location |> pathname) {
-    | ""
-    | "/" => []
-    | raw =>
-      /* remove the preceeding /, which every pathname seems to have */
-      let raw = Js.String.sliceToEnd(~from=1, raw);
-      /* remove the trailing /, which some pathnames might have. Ugh */
-      let raw =
-        switch (Js.String.get(raw, Js.String.length(raw) - 1)) {
-        | "/" => Js.String.slice(~from=0, ~to_=-1, raw)
-        | _ => raw
-        };
-      raw |> Js.String.split("/") |> arrayToList;
-    }
+let hsMatcher = [%re "/(#[^?&]+)?([?&].*)?$/"];
+let pathMatcher = [%re "/(#|\\?).+$/"];
```

Comment:
Can we just split on `?` and send these separate ways? Did you write these regex from scratch or get them from somewhere?

Reply:
I wrote them from scratch, we can split on ?since we're not dealing with hash anymore; My idea was that since I can trust a string I wanted to account for the possibility of things like hash and search being mixed in, I will ammed your comments and we can take another look.